pyproject.toml

```diff
@@ -31,6 +31,7 @@ dependencies = [
     "rich>=12.0,<14.0",
     "rope>=0.21,<2.0",  # subject to LGPLv3 license
     "toml>=0.10.0",
+    "typing_extensions>=4.0",
```

Comment:
`Self` type is introduced in Python 3.11. `typing_extensions` contains backports of newly-introduced types to previous Python versions. It is maintained by core Python developers like Guido and Jelle Zijlstra and doesn't introduce any additional dependencies. Once we drop Python<3.11, it can be dropped.

Reply:
Ah makes sense.